crates/cxx-qt-lib-headers/include/qset.h

```diff
@@ -25,11 +25,16 @@ struct rust::IsRelocatable<QSet<T>> : std::true_type
 namespace rust {
 namespace cxxqtlib1 {
 
+template<typename T>
+::rust::isize
+qsetLen(const QSet<T>& s) noexcept;
+
 template<typename T>
 const T&
-qsetGetUnchecked(const QSet<T>& s, ::std::size_t pos) noexcept
+qsetGetUnchecked(const QSet<T>& s, ::rust::isize pos) noexcept
 {
-  Q_ASSERT(pos < static_cast<::std::size_t>(s.size()));
+  Q_ASSERT(pos < qsetLen(s));
+  Q_ASSERT(pos >= 0);
   auto it = s.cbegin();
   std::advance(it, pos);
```

Comment:
Qt uses effectively `return s.cbegin()[pos];` internally, is that more performant / better than using `std::advance` ?

Reply:
Looks like `std::advance` is safe / more correct as depending how the container is organised simply using an index offset might not be the Nth item, whereas `std::advance` can take this into account - although it's slower potentially.